---
name: "Stephen M"
surname: "Pederson"
position: "Post-Doctoral Bioinformatician"
address: |
  | Dame Roma Mitchell Cancer Research Laboratories
  | Adelaide Medical School
  | University of Adelaide
  |
phone: +61 4 1333 9618
email: "stephen.pederson@adelaide.edu.au"
www: "researchers.adelaide.edu.au/profile/stephen.pederson"
github: steveped
linkedin: "stephen-pederson-06ba4914"
#orcid: 0000-0001-8197-3303
#twitter: StevePederson8
date: "`r format(Sys.time(), '%B %Y')`"
output: 
  vitae::moderncv:
    keep_tex: true
    theme: classic
---

<!-- Line spacing from detailed_entries() is too much. Reset here -->
\renewcommand*{\cventry}[7][-3.5mm]{%
  \cvitem[#1]{#2}{%
    {\bfseries#3}%
    \ifthenelse{\equal{#4}{}}{}{, {\slshape#4}}%
    \ifthenelse{\equal{#5}{}}{}{, #5}%
    \ifthenelse{\equal{#6}{}}{}{, #6}%
    .\strut%
    \ifx&#7&%
    \else{\newline{}\begin{minipage}[t]{\linewidth}\small#7\end{minipage}}\fi}}


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
```

```{r packages}
library(vitae)
library(tidyverse)
library(magrittr)
library(scholar)
library(lubridate)
library(scales)
```


```{r getPubs}
sp <- "ZbO2KUEAAAAJ"
pubsRaw <- get_publications(sp, flush = TRUE) %>%
  as_tibble() %>%
  mutate(
    across(
      .cols = c("title", "author", "journal"), 
      .fns = stringi::stri_trans_general,
      id = "latin-ascii"
    )
  ) %>%
  dplyr::filter(
    !str_detect(title, "BMEA")
  )
```

```{r getIF}
j <- unique(pubsRaw$journal) %>%
  .[. != ""] %>%
  str_subset("xiv$", negate = TRUE) %>%
  str_subset("Patent", negate = TRUE)
imp <- get_impactfactor(j) %>%
  mutate(
    j = j,
    ImpactFactor = case_when(
      # All taken from https://academic-accelerator.com/Impact-Factor-IF
      j %in% c("Plutella", "The Journal of Open Source Software") ~ NA_real_,
      j == "Oncotarget" ~ 3.71,
      j == "Epigenomes" ~ 3.83,
      j == "The Journal of Immunology" ~ 4.34,
      j == "Clinical & Translational Immunology" ~ 9.18,
      j == "International Journal of Nephrology and Renovascular Disease" ~ 2.59,
      !is.na(ImpactFactor) ~ ImpactFactor
    )
  )
pubsRaw %<>%
  left_join(
    dplyr::select(imp, journal = j, ImpactFactor),
    by = "journal"
  )
```

```{r shorten_full_authors}
shorten_full_authors <- function(x){
  x <- iconv(x, "latin1", "ASCII", "n")
  x <- stringi::stri_trans_general(x, id = "latin-ascii")
  x <- str_split(x, pattern = ",")
  x <- lapply(x, str_replace_all, pattern = "[a-z]+ ", replacement = " ")
  x <- lapply(x, str_trim)
  x <- lapply(x, str_replace_all, pattern = "([A-Z]) ([A-Z]) (.+)", replacement = "\\1\\2 \\3")
  vapply(x, str_flatten, character(1), collapse = ", ")
}
```

```{r replace_short, results='hide'}
# needsFull <- !str_detect(pubsRaw$author, "Pederson")
# Can't be more than 50
needsFull <- str_detect(pubsRaw$author, "\\.{3}$")
full <- get_complete_authors(sp, pubsRaw$pubid[needsFull], initials = FALSE)
pubsRaw$author[needsFull] <- shorten_full_authors(full)
## Correct the weird ones
pubsRaw$author <- str_replace_all(pubsRaw$author, "(SC B )(T Sadlon.+)", "\\2, SC Barry")
pubsRaw$author[str_detect(pubsRaw$author, "RJ$")] <- full %>%
  str_subset("Gonda") %>% 
  str_replace("Gonda TJ", "Gonda TJ,") %>% 
  str_split(pattern = ",") %>% 
  .[[1]] %>% 
  str_trim() %>% 
  str_replace_all("(.+) (.+)", "\\2 \\1") %>% 
  str_flatten(collapse = ", ")
```


```{r pred_h}
pred_h <- predict_h_index(sp)
```

```{r finalPubs}
pubs <- pubsRaw %>%
  dplyr::filter(
    # cites > 1 | year == "2020", #year(today()),
    !str_detect(journal, "(Patent|EUROPEAN|REPRODUCTIVE|Wiley)")
  ) %>%
  mutate(
    author = str_replace_all(author, "\\.\\.\\.", "\\\\emph{et al}"),
    author = str_replace_all(author, " &", " \\&"),
    title = str_replace_all(title, " &", " \\&"),
    title = case_when(
      title == "Brain Transcriptome Analysis of a Protein-Truncating Mutation in Sortilin-Related Receptor 1 Associated With Early-Onset Familial Alzheimer's Disease Indicates Early Effects on ..." ~ str_replace(title, "\\.{3}", "Mitochondrial and Ribosome Function"),
      title == "Brain transcriptome analysis reveals subtle effects on mitochondrial function and iron homeostasis of mutations in the SORL1 gene implicated in early onset familial Alzheimer's ..." ~ str_replace(title, "\\.{3}", "Disease"),
      title == "Transcriptome analyses of 7-day-old zebrafish larvae possessing a familial Alzheimer's disease-like mutation in psen1 indicate effects on oxidative phosphorylation, mcm ..." ~ str_replace(title, "\\.{3}", "functions, and iron homeostasis"),
      TRUE ~ title
    ),
    journal = str_replace_all(journal, " &", " \\\\&"),
    journal = str_to_title(journal),
    journal = str_replace_all(journal, "^Bmc", "BMC"),
    journal = str_replace_all(journal, "^Plos", "PLOS"),
    journal = str_replace_all(journal, "^Mbio", "mBio"),
    journal = str_replace_all(journal, "Dna$", "DNA"),
    journal = str_replace_all(journal, "Biorxiv", "bioRxiv"),
    journal = str_replace_all(journal, "Medrxiv", "medRxiv"),
    journal = case_when(
      grepl("BMEA", title) ~ "PhD Thesis",
      !grepl("BMEA", title) ~ journal
    )
  ) 
fl <- pubs %>%
  dplyr::filter(
    str_detect(author, "Pederson$") |
      str_detect(title, "Transposable elements .+ evolution of amniotes")
    ) %>%
  pull(pubid)
wif <- weighted.mean(pubs$ImpactFactor, pubs$cites, na.rm = TRUE)
```

```{r}
q1 <- c(
  "Atherosclerosis",
  "Bioinformatics",
  "Blood Cancer Journal",
  "Clinical \\& Translational Immunology",
  "Frontiers In Immunology",
  "Journal Of Experimental Medicine",
  "mBio",
  "Molecular Brain",
  "Molecular Ecology",
  "Neurobiology Of Disease",
  "Oncogene",
  "Placenta",
  "Science Of The Total Environment",
  "Scientific Reports"
)
q2 <- c(
 "Behavioural Brain Research",
 "Brain Research Bulletin",
 "European Journal Of Immunology",
 "Immunology And Cell Biology",
 "Journal Of Alzheimer's Disease",
 "Mobile DNA",
 "PLOS One",
 "The Journal Of Immunology"
) %>%
  setdiff(q1)
q_perc <- pubs %>%
  dplyr::filter(
    !str_detect(journal, "xiv$") # Currently in review
  ) %>%
  summarise(
    q1 = mean(journal %in% q1),
    q2 = mean(journal %in% q2)
  )
```


```{r pub2Tex}
pub2Tex <- function(x, .fl = fl){
  lab <- ifelse(x$pubid %in% .fl, "\\moditem{*}", "\\item")
  paste(
    "\t",
    lab,
    # "\\item",
    str_replace_all(x$author, "SM* Pederson", "\\\\textbf{SM Pederson}") %>%
      str_replace_all("Pederson SM*", "\\\\textbf{Pederson SM}"),
      # str_replace_all("‚Äê", "-"),
    paste0("(", x$year, ")"),
    paste0(
      "\\emph{", x$title, "}"
    ),
    paste0(
      "\\textbf{", x$journal, "}"
    ),
    x$number,
    # paste0(
    #   "(\\emph{",
    #   x$cites,
    #   " citations})"
    # ),
    "\\\\[-4mm]",
    "\n"
  ) 
}
```


# Profile

 * A keen interest of mine is *Transcriptomics and Gene Regulatory Networks*. As part of my PhD (2018), whilst training in the context of the immune dysregulation of T cells, I developed a novel, Bayesian statistical methodology for the analysis of Whole Transcript Microarray data, focussing on the detection of alternate isoform usage. As a consequence, a potential role for ncRNA in the activation of T cells was revealed for the first time.
 * As the inaugural co-ordinator of the Bioinformatics Hub at the University of Adelaide (2014-2020), *I oversaw the growth of this facility from a sole employee in 2014 to 4 full-time core staff by 2020*, along with a further 12 contract and casual staff. Collectively, our team were named investigators on $8.5m in research funding across all Faculties, and gaining experience in a diversity of high-throughput data types.
 * Bioinformatics education and training have always been an important interest. 
 In 2015 I attended the *Master R Developer Workshop* with Dr Hadley Wickham (RStudio).
 Locally *I have delivered bioinformatics training workshops to over 1500 staff and students*, as well as delivering training  at national events. From the years 2016-2018, I ran the *largest training event in the language `R` held in Australia*, attracting attendees from around the country, and have also run corporate R training for both Sugar Research Australia (QLD) and SARDI
 * I played an active role in the *establishment of an Undergraduate Major in Bioinformatics* (B.Sc), acting as Course Convenor for Transcriptomics Applications III in 2020, writing and delivering nearly all lecture and practical material. 
 * I am an enthusiastic supporter of open source software and am the senior author/developer of *two Bioconductor packages*. From 2016-2019, I played an active role in establishing the international event BiocAsia, bringing the training and networking opportunities of the Bioconductor community to our local region. 
 <!-- * Having published `r nrow(dplyr::filter(pubs, !str_detect(journal, "iv$")))` peer-reviewed papers, the average Impact Factor across all my accepted publications is currently `r round(wif, 2)` with over 500 citations. Despite being an ECR of 2 years, I have an *h-index* of `r pred_h$h_index[1]`. -->
 <!-- * In addition to bioinformatics, I am active in the local music community, being a Musical Director for Adelaide Cabaret Festival (Brenta's Bad Thing, 2014), and making multiple appearances at both the Adelaide Guitar Festival and WOMADelaide. -->

\small

# Employment History

```{r}
tribble(
  ~Start, ~End, ~Org, ~Role,
  1992, 2014, "", "Self-Employed Musician",
  2011, 2014, "School of Mathematical Sciences, University of Adelaide", "Casual Tutor",
  2014, 2020, "Bioinformatics Hub, University of Adelaide", "Co-ordinator",
  2020, NA_real_, "Dame Roma Mitchell Cancer Research Laboratories, Univerrsity of Adelaide", "Post-Doctoral Bioinformatician",
) %>% 
  arrange(desc(Start)) %>%
  mutate(
    Year = case_when(
      Start == End ~ as.character(Start),
      is.na(End) ~ paste0(Start,"-"),
      Start != End ~ paste(Start, End, sep = "-")
    ),
    Loc = "Adelaide, South Australia"
  ) %>%
  detailed_entries(
    Role, Year, Org, where = Loc
  )
```


# Education

```{r}
tribble(
  ~ Degree, ~ Year, ~ Institution, ~ Where,
  "Bachelor of Music", "1989-1991", "University of Adelaide", "South Australia, Australia",
  "Bachelor of Science", "2002-2005", "University of Adelaide", "South Australia, Australia",
  "Bachelor of Mathematics & Comp. Sc. (Hons)", "2006-2007", "University of Adelaide", "South Australia, Australia",
  "Doctor of Philosophy (Medicine)", "2008-2018", "University of Adelaide", "South Australia, Australia"
) %>% 
  arrange(desc(Year)) %>%
  detailed_entries(Degree, Year, Institution, Where)
```


# Presentations

```{r}
tribble(
  ~Year, ~Event, ~Location, ~Title, 
  2019, "Bioconductor", "New York, NY", "Short Talk & Poster - ngsReports and strandCheckR: Two new packages for quality control of NGS data",
  2019, "BiocAsia/BioinfoSummer", "Sydney", "Short Talk - Experiences of a First-Time Package Submitter",
  2017, "BioinfoSummer", "Adelaide", "Training Workshop - Data Visualisation in R",
  2016, "ABACBS", "Brisbane", "Poster - Use of a weighted bootstrap for indentifying the regulatory role of genomic repeat elements",
  2015, "NCBI", "Bethesda, MD", "Presentation - Bioinformatics from Down Under",
  2015, "BioinfoSummer", "Sydney", "Training Workshop - Introduction to R"
) %>%
  arrange(desc(Year)) %>%
  detailed_entries(
    Event,
    Year,
    Location,
    Title
  )
```


# Community Engagement

```{r}
tribble(
  ~Start, ~End, ~Org, ~Role,
  2018, 2021, "Australian Bioinformatics And Computational Biology Society", "Ordinary Member",
  2016, 2019, "Bioconductor", "BiocAsia Conference Committee",
  2017, 2017, "Australian Bioinformatics And Computational Biology Society", "Conference Committee",
  2016, 2016, "Australian Mathematical Sciences Institute", "BioinfoSummer Conference Committee",
  2014, 2016, "Australasian Genomics Technologies Association", "Ordinary Member",
  2014, 2015, "International Society for Computational Biology", "Ordinary Member"
) %>% 
  arrange(desc(End)) %>%
  mutate(
    Year = case_when(
      Start == End ~ as.character(Start),
      Start != End ~ paste(Start, End, sep = "-")
    )
  ) %>%
  brief_entries(
    Role,
    Year, 
    Org
  )
```

# Technical Skills

 * Highly fluent in the language `R` and using High-Performance Computing systems
 * Experienced with `linux` (Ubuntu), `bash`, `Git` for version control, `markdown` and \LaTeX, along with `snakemake` for HPC pipeline management
 * Expertise in statistics, RNA-Seq, scRNA-Seq, ChIP-Seq, RAD/GBS-Seq, HiC-Seq and microarray data
 * A working familiarity with `python` and `C/C++`

```{r yr}
yr <- 2018
```


# Publications Since `r yr` (`r nrow(pubs)` in total since `r min(pubs$year)`)

`r pubs %>% dplyr::filter(year >= yr, !str_detect(journal, "xiv$")) %>% summarise(q1 = mean(journal %in% q1)) %>% pull("q1") %>% percent()` of the below are in Q1 journals. First or senior author publications are indicated with an asterisk.  

\vspace{4mm}

\begin{modenumerate}
  \small
```{r, results='asis'}
printPubs <- pubs %>%
  arrange(desc(year)) %>%
  dplyr::filter(year >= yr) %>%
  split(f = seq_len(nrow(.))) %>%
  lapply(pub2Tex)
for (i in seq_along(printPubs)) cat(printPubs[[i]])
```

\end{modenumerate}

```{r, eval=FALSE}
# Just doesn't look as nice as my manual version
bibliography_entries("scholar.bib") %>%
  arrange(issued, desc(author$family)) %>%
  dplyr::filter(
    !is.na(`container-title`),
    !str_detect(`container-title`, "EUROPEAN"),
    !type %in% c("no-type", "paper-conference")
  )
```

